src/giskardpy/tfwrapper.py

```diff
@@ -152,6 +171,104 @@ def lookup_point(target_frame, source_frame, time=None):
     return p
 
 
+def euler_to_quaternion(rpy):
+    """
+    :type rpy: List[double]
+    :rtype: Quaternion
+    """
+    q = quaternion_from_matrix(euler_matrix(rpy[0], rpy[1], rpy[2]))
+    return Quaternion(q[0], q[1], q[2], q[3])
+
+
+def tf_list_to_matrix(tf_list, tf_list_euler=False):
+    """
+    :type tf_list: List[List[double], List[double]]
+    :rtype: 4x4 Matrix
+    """
+    tf_trans_mat = translation_matrix([tf_list[0][0],
+                                       tf_list[0][1],
+                                       tf_list[0][2]])
+    if tf_list_euler:
+        tf_rot_mat = euler_matrix(tf_list[1][0],
+                                  tf_list[1][1],
+                                  tf_list[1][2])
+    else:
+        tf_rot_mat = quaternion_matrix(tf_list[1])
+    return np.dot(tf_trans_mat, tf_rot_mat)
+
+
+def multiply_transform_lists(tf_list, other_tf_list, tf_list_euler=False, other_tf_list_euler=False):
+    """
+    :type tf_list: List[List[double], List[double]]
+    :type other_tf_list: List[List[double], List[double]]
+    :rtype: List[List[double], List[double]]
+    """
+    tf_mat = tf_list_to_matrix(tf_list, tf_list_euler=tf_list_euler)
+    other_tf_mat = tf_list_to_matrix(other_tf_list, tf_list_euler=other_tf_list_euler)
+    new_tf = np.dot(tf_mat, other_tf_mat)
+    t = translation_from_matrix(new_tf)
+    q = quaternion_from_matrix(new_tf)
+    return [t, q]
+
+
+def transform_to_list(tf):
+    """
+    :type tf: Transform
+    :rtype: List[List[double], List[double]]
+    """
+    ctf = copy(tf)
+    return [[ctf.translation.x, ctf.translation.y, ctf.translation.z],
+            [ctf.rotation.x, ctf.rotation.y, ctf.rotation.z, ctf.rotation.w]]
+
+
+def list_to_transform(l, list_euler=False):
+    """
+    :type l: List[List[double], List[double]]
+    :rtype: Transform
+    """
+    cl = copy(l)
+    return Transform(Vector3(cl[0][0], cl[0][1], cl[0][2]),
+                     Quaternion(cl[1][0], cl[1][1], cl[1][2], cl[1][2])
+                     if not list_euler else euler_to_quaternion(cl[1]))
+
+
+def list_to_pose(l, list_euler=False):
+    """
+    :type l: List[List[double], List[double]]
+    :rtype: Pose
+    """
+    cl = copy(l)
+    return Pose(Point(cl[0][0], cl[0][1], cl[0][2]),
+                Quaternion(cl[1][0], cl[1][1], cl[1][2], cl[1][2])
+                if not list_euler else euler_to_quaternion(cl[1]))
```

Comment:
I don't think there is a need for these list related functions. This is very urdf specific, which is why I wrote in another comment, that the URDFObject should handle returning ros messages. Maybe they can be moved into that file.